src/charm.py

```diff
@@ -487,66 +483,109 @@ def _hwcksum_config_matches_pod_config(self) -> bool:
             existing_content = {}
         return existing_content.get("hwcksum") != self._charm_config.enable_hw_checksum
 
+    def _unit_is_non_active_status(self):
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to perform if we're removing the
+            # charm.
+            logger.info("Scaling is not implemented for this charm")
+            return BlockedStatus("Scaling is not implemented for this charm")
+        try:  # workaround for https://github.com/canonical/operator/issues/736
+            self._charm_config: CharmConfig = CharmConfig.from_charm(charm=self)  # type: ignore[no-redef]  # noqa: E501
+        except CharmConfigInvalidError as exc:
+            logger.info(exc.msg)
+            return BlockedStatus(exc.msg)
+        if not self._is_cpu_compatible():
+            return BlockedStatus("CPU is not compatible, see logs for more details")
+        if not self._kubernetes_multus.multus_is_available():
+            logger.info("Multus is not installed or enabled")
+            return BlockedStatus("Multus is not installed or enabled")
+        if not self._bessd_container.can_connect():
+            logger.info("Waiting for bessd container to be ready")
+            return WaitingStatus("Waiting for bessd container to be ready")
+        if not self._kubernetes_multus.is_ready():
+            logger.info("Waiting for Multus to be ready")
+            return WaitingStatus("Waiting for Multus to be ready")
+        if not self._bessd_container.exists(path=BESSD_CONTAINER_CONFIG_PATH):
+            logger.info("Waiting for storage to be attached")
+            return WaitingStatus("Waiting for storage to be attached")
+        if container_status := self._container_status_setter():
+            return container_status
+
+    def _container_status_setter(self):
```

Comment:
This method lacks return typing plus it's name could be better. It is not a setter.